### PR TITLE
[heft] Ensure Heft returns a nonzero exit code when errors are reported

### PR DESCRIPTION
**Pull request by octogonz**
Eventually we should redesign how `process.exitCode` is managed by Heft and Rush. This PR is a quick fix.

### CODE REVIEW

File: common/reviews/api/node-core-library.api.md [hunk 1]
```diff
@@ -650,5 +657,11 @@ export enum TextAttribute {
     Underline = 2
 }
 
+// @public
+export class UuidTypeId {
+    static isInstanceOf(targetObject: unknown, typeUuid: string): boolean;
+    static registerClass(targetClass: new (...args: any[]) => any, typeUuid: string): void;
+    }
```
Comment (iclanton) [inline common/reviews/api/node-core-library.api.md]: Want to use this in [`FileError`](https://github.com/microsoft/rushstack/blob/master/apps/heft/src/pluginFramework/logging/FileError.ts) as well?

File: libraries/node-core-library/src/TypeUuid.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+ * import { Widget as Widget1 } from 'v1-of-library';
+ * import { Widget as Widget2 } from 'v2-of-library';
+ * const widget = new Widget2();
+ * console.log(widget instanceof Widget1); // prints true
+ * ```
+ *
+ * @public
+ */
+export class TypeUuid {
+  private static _uuidRegExp: RegExp = /^[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}$/;
```
Comment (iclanton) [inline libraries/node-core-library/src/TypeUuid.ts:42]: This kinda seems like an unnecessary constraint.

File: libraries/node-core-library/src/TypeUuid.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+   */
+  public static isInstanceOf(targetObject: unknown, typeUuid: string): boolean {
+    let objectPrototype: {} = Object.getPrototypeOf(targetObject);
+    while (objectPrototype !== undefined && objectPrototype !== null) {
+      const registeredUuid: string = objectPrototype[classPrototypeUuidSymbol];
+      if (registeredUuid !== undefined) {
+        if (registeredUuid === typeUuid) {
+          return true;
+        }
+      }
```
Comment (iclanton) [inline libraries/node-core-library/src/TypeUuid.ts:78]:
```suggestion
      if (registeredUuid === typeUuid) {
        return true;
      }
```

File: libraries/node-core-library/src/TypeUuid.ts (hunk 2 above)
Comment (octogonz) [reply libraries/node-core-library/src/TypeUuid.ts:42]: We can relax it later if there is a good reason. I want to be explicit about the format.

File: libraries/node-core-library/src/TypeUuid.ts (hunk 3 above)
Comment (octogonz) [reply libraries/node-core-library/src/TypeUuid.ts:78]: Fixed

File: common/reviews/api/node-core-library.api.md (hunk 1 above)
Comment (octogonz) [reply common/reviews/api/node-core-library.api.md]: Good idea, fixed

**Pull request merged**